Api/V8/Service/ModuleService.php

```diff
@@ -113,39 +116,92 @@ public function getRecords(GetModulesParams $params, Request $request)
         }
 
-        $beanListResponse = $this->beanManager->getList($module)
-            ->orderBy($orderBy)
-            ->where($where)
-            ->offset($offset)
-            ->limit($limit)
-            ->max($size)
-            ->deleted($deleted)
-            ->fields($this->beanManager->filterAcceptanceFields($bean, $fields))
-            ->fetch();
-
-
-        $beanArray = [];
-        foreach ($beanListResponse->getBeans() as $bean) {
-            $bean = $this->beanManager->getBeanSafe(
-                $params->getModuleName(),
-                $bean->id
-            );
-            $beanArray[] = $bean;
+
+        // Detect if bean has email field
+        if ((property_exists($bean, 'email1') && strpos($where, 'email1')) || (property_exists($bean,
+                    'email2') && strpos($where, 'email2'))) {
+            $selectedFields = strtolower($module) . '.' . implode(',' . substr(strtolower($module), 0, 1) . '.', $fields);
+            $selectedModule = strtolower($module);
+
+            //Selects Module or COUNT(*) and will add one to the query.
+            $idSelect = "SELECT {$selectedModule}.id ";
+            $countSelect = 'SELECT COUNT(*) AS cnt ';
+
+            //Email where clause
+            $fromQuery = "FROM email_addresses join email_addr_bean_rel on email_addresses.id = email_addr_bean_rel.email_address_id join {$selectedModule} on {$selectedModule}.id = email_addr_bean_rel.bean_id ";
+            $modifiedWhere = str_replace("accounts.email1", "email_addresses.email_address", $where);
+            $where = "$modifiedWhere";
+
+            //Sets and adds deleted to the query
+            if ($deleted == 0) {
+                $where_auto = "$bean->table_name.deleted=0";
+            } elseif ($deleted == 1) {
+                $where_auto = "$bean->table_name.deleted=1";
+            }
+            if ($where != "") {
+                $where = " where ($where) AND $where_auto";
+            } else {
+                $where = " where $where_auto";
+            }
+
+            //Joins parts together to form query
+            $query = $idSelect . $fromQuery . $where;
+            $countQuery = $countSelect . $fromQuery . $where;
+            $realRowCount = (int)$db->fetchRow($db->query($countQuery, true, ""))["cnt"];;
+
+            //Sets orderby into the query
+            $order_by = $bean->process_order_by($orderBy);
+            if (!empty($orderBy)) {
+                $query .= ' ORDER BY ' . $order_by;
+            }
+
+            $result = $bean->process_list_query($query, $offset, $limit, -1, $where);
+            $beanResult['row_count'] = $result['row_count'];
+            $beanList = [];
+
+            foreach($result['list'] as $resultBean)
+            {
+                $queryModuleBean = \BeanFactory::newBean($module);
+                $queryModuleBean->id = $resultBean->id;
+                array_push($beanList, $queryModuleBean);
```

Comment:
Technically a micro optimisation but this is 2x faster and imo looks a lot nicer :).

```suggestion
                $beanList[] = $queryModuleBean;
```